earthmover/runs_file.py

```diff
@@ -157,10 +158,15 @@ def _build_hashes(self) -> Dict[str, str]:
 
             if f"$sources.{source.name}" not in node_data.keys():
                 continue
+                
+            source_classname = source.__class__.__name__
 
-            if not source.is_remote and source.file and not os.path.isdir(source.file):
+            if source.is_hashable and source_classname=="FileSource" and source.file and not os.path.isdir(source.file):
```

Comment:
I'd argue that the get_hash method should be defined within the Source subclass instead of the class type being referenced in the runs file. Then we can just do the following here:
```
if source.is_hashable:
    sources_hash += source.get_hash()
```

We can use polymorphism to call the same interface across sources without the runs file needing to know which source type is being called.

Reply:
Great suggestion, thanks @jayckaiser. I refactored this as you suggest, with the slight added complexity that, because `runs_file.py` also used `_get_file_hash()` to compute file hashes for templates and `map_values` operation files, I moved `get_file_hash()` to `util.py` so that both `FileSource.get_hash()` and `runs_file.py` can both call it.

Let me know if you have any other changes to suggest.